common/persistence/tests/queue_v2_test_suite.go

```diff
@@ -412,3 +413,120 @@ func RunQueueV2TestSuiteForSQL(t *testing.T, factory *sql.Factory) {
 		tests.RunSQLQueueV2TestSuite(t, db)
 	})
 }
+
+func testListQueues(ctx context.Context, t *testing.T, queue persistence.QueueV2) {
+	t.Run("HappyPath", func(t *testing.T) {
+		// ListQueues when empty
+		queueType := persistence.QueueTypeHistoryDLQ
+		response, err := queue.ListQueues(ctx, &persistence.InternalListQueuesRequest{
+			QueueType:     queueType,
+			PageSize:      10,
+			NextPageToken: nil,
+		})
+		require.NoError(t, err)
+		require.Equal(t, 0, len(response.QueueNames))
+
+		// List one queue.
+		queueName := "test-queue-" + t.Name() + "first"
+		_, err = queue.CreateQueue(ctx, &persistence.InternalCreateQueueRequest{
+			QueueType: queueType,
+			QueueName: queueName,
+		})
+		require.NoError(t, err)
+		response, err = queue.ListQueues(ctx, &persistence.InternalListQueuesRequest{
+			QueueType:     queueType,
+			PageSize:      10,
+			NextPageToken: nil,
+		})
+		require.NoError(t, err)
+		require.Equal(t, 1, len(response.QueueNames))
+		require.Equal(t, queueName, response.QueueNames[0])
+
+		// List multiple queues.
+		queueName = "test-queue-" + t.Name() + "second"
+		_, err = queue.CreateQueue(ctx, &persistence.InternalCreateQueueRequest{
+			QueueType: queueType,
+			QueueName: queueName,
+		})
+		require.NoError(t, err)
+		response, err = queue.ListQueues(ctx, &persistence.InternalListQueuesRequest{
+			QueueType:     queueType,
+			PageSize:      10,
+			NextPageToken: nil,
+		})
+		require.NoError(t, err)
+		require.Equal(t, 2, len(response.QueueNames))
+		require.Contains(t, response.QueueNames, queueName)
+
+		// List multiple queues in pages.
+		var queueNames []string
+		for i := 0; i < 3; i++ {
+			queueNames = append(queueNames, "test-queue-"+t.Name()+strconv.Itoa(i))
+		}
+		for _, queueName := range queueNames {
+			_, err := queue.CreateQueue(ctx, &persistence.InternalCreateQueueRequest{
+				QueueType: queueType,
+				QueueName: queueName,
+			})
+			require.NoError(t, err)
+		}
+		var listedQueueNames []string
+		response, err = queue.ListQueues(ctx, &persistence.InternalListQueuesRequest{
+			QueueType:     queueType,
+			PageSize:      1,
+			NextPageToken: nil,
+		})
+		require.NoError(t, err)
+		require.Equal(t, 1, len(response.QueueNames))
+		listedQueueNames = append(listedQueueNames, response.QueueNames...)
+		response, err = queue.ListQueues(ctx, &persistence.InternalListQueuesRequest{
+			QueueType:     queueType,
+			PageSize:      1,
+			NextPageToken: response.NextPageToken,
+		})
+		require.NoError(t, err)
+		require.Equal(t, 1, len(response.QueueNames))
+		listedQueueNames = append(listedQueueNames, response.QueueNames...)
+		response, err = queue.ListQueues(ctx, &persistence.InternalListQueuesRequest{
```

Comment:
I think we need to check the listedQueueNames here (e.g. the .Contains loop).

Reply:
There is a contained loop at the end of the test. https://github.com/temporalio/temporal/pull/5098/files/157f0d0f63456ed33a9e45fc81b8b37455c9778d#diff-c932021410a4891568e8793de8c80f7728b87fd72ec27bcd808f751ec703f271R507

After all paginated list requests, this loop checks if all the queues are returned.
It is not possible to have this test in between list requests as we don't have any guarantees for order in which queues will be returned.